src/main.cpp

```diff
@@ -12,47 +13,83 @@ void help() {
                  "  --immediate              - Lock immediately, ignoring any configured grace period\n"
                  "  -h, --help               - Show this help message\n";
 }
+
+std::optional<std::string> parseArg(const std::vector<std::string>& args, const std::string& flag, std::size_t& i) {
+    if (i + 1 < args.size()) {
+        return args[++i];
+    } else {
+        std::cerr << "Error: Missing value for " << flag << " option.\n";
+        return std::nullopt;
+    }
+}
+
 int main(int argc, char** argv, char** envp) {
     std::string configPath;
     std::string wlDisplay;
     bool        immediate = false;
+    bool        showHelp = false;
+
+    std::vector<std::string> args(argv, argv + argc);
 
-    for (int i = 1; i < argc; ++i) {
-        std::string arg = argv[i];
+    for (std::size_t i = 1; i < args.size(); ++i) {
+        const std::string arg = argv[i];
 
         if (arg == "--verbose" || arg == "-v")
             Debug::verbose = true;
 
         else if (arg == "--quiet" || arg == "-q")
             Debug::quiet = true;
 
-        else if ((arg == "--config" || arg == "-c") && i + 1 < argc)
-            configPath = argv[++i];
+        else if ((arg == "--config" || arg == "-c") && i + 1 < static_cast<std::size_t>(argc))
+            if (auto value = parseArg(args, arg, i); value) {
+                configPath = *value;
+            } else {
+                return 1;
+            }
 
-        else if (arg == "--display" && i + 1 < argc) {
-            wlDisplay = argv[i + 1];
-            i++;
+        else if (arg == "--display" && i + 1 < static_cast<std::size_t>(argc)) {
```

Comment:
why not just use C-style casts? They are shorter.

Reply:
I just default to static_cast since they can catch wrong conversions during compile time, but I don't see need for that in this case. I can change it to C-style if you want.

Reply:
yeah we usually use c style